pulsar/consumer_impl.go

```diff
@@ -459,7 +459,9 @@ func (c *consumer) AckID(msgID MessageID) error {
 	}
 
 	if mid.consumer != nil {
-		return mid.Ack()
+		if pc, ok := (mid.consumer).(*partitionConsumer); ok && pc.getConsumerState() == consumerReady {
+			return mid.Ack()
```

Comment:
Thanks @billowqiu work for this, can we abstract a public method to check if the consumer object is active?